### ISSUE TITLE
Demos include a 'Dismiss' semantics object when not needed

### ISSUE THREAD

**Issue by guidezpl** [flutter/gallery#209]
A side-effect of https://github.com/flutter/gallery/commit/63eb33e62c9847716119cfb9c6594166ec835304 is that all demo pages now includes a `Semantics` object to dismiss an open menu, even when that menu is closed.

This should be fixed, while still retaining demo state (conditionally inserting a Semantics widget caused a rebuild that lost demo state)
